modules/quadbin/bigquery/sql/QUADBIN_TOPARENT.sql

```diff
@@ -0,0 +1,13 @@
+----------------------------
+-- Copyright (C) 2022 CARTO
+----------------------------
+
+CREATE OR REPLACE FUNCTION `@@BQ_PREFIX@@carto.QUADBIN_TOPARENT`
+(quadbin INT64, resolution INT64)
+RETURNS INT64
+AS ((
+    IF(resolution < 0 OR resolution > ((quadbin >> 52) & 0x1F),
```

Comment:
Should resolution 0 be allowed?

Reply:
Also `resolution >= ((quadbin >> 52) & 0x1F)`?
To avoid creating parents at the same resolution of the input

Reply:
> Should we block ((quadbin >> 52) & 0x1F) < 1?

Not really, a quadbin res0 can compute its parent res0, which is the same index

> Also resolution >= ((quadbin >> 52) & 0x1F)?
> To avoid creating parents at the same resolution of the input

Not a problem computing a parent at resolution 0, by definition is the same index